navit/navit.c

```diff
@@ -3504,6 +3504,27 @@ void navit_layout_switch(struct navit *n) {
                 && valid_attr.u.num==attr_position_valid_invalid) {
             return; //No valid fix yet
         }
+
```

Comment:
Suggest to move this check below line 3531. The the check for fix is done before that.